x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/policy_template_input_selector.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import {
+  EuiFlexGroup,
+  EuiToolTip,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { css } from '@emotion/react';
+import type { PostureInput, PosturePolicyTemplate } from '../../../common/constants';
+import { getPolicyTemplateInputOptions, type NewPackagePolicyPostureInput } from './utils';
+import { InlineRadioGroup } from './inline_radio_group';
+
+type Props = { input: NewPackagePolicyPostureInput } & (
+  | { disabled: true }
+  | {
+      disabled?: false;
+      updatePolicyInput(id: string): void;
+    }
+);
+
+const noop = () => {};
+
+const RadioLabel = ({
+  label,
+  icon,
+  disabled,
+  tooltip,
+}: ReturnType<typeof getPolicyTemplateInputOptions>[number]) => (
+  <EuiToolTip content={tooltip} anchorProps={{ style: { width: '100%' } }}>
+    <EuiFlexGroup direction="row" alignItems="center">
+      <EuiFlexItem grow={true}>{label}</EuiFlexItem>
+      {icon && (
+        <EuiFlexItem grow={false}>
+          <EuiIcon
+            type={icon}
+            css={
+              disabled &&
+              css`
+                filter: grayscale(1);
+              `
+            }
+          />
+        </EuiFlexItem>
+      )}
+    </EuiFlexGroup>
+  </EuiToolTip>
+);
+
+export const PolicyInputSelector = ({ input, ...rest }: Props) => {
+  const baseOptions = getPolicyTemplateInputOptions(input.policy_template as PosturePolicyTemplate);
+  const options = baseOptions.map((option) => ({
+    ...option,
+    disabled: option.disabled || rest.disabled,
+    label: <RadioLabel {...option} />,
+  }));
+
+  return (
+    <div>
+      <InlineRadioGroup
+        idSelected={input.type}
+        options={options}
+        onChange={rest.disabled ? noop : rest.updatePolicyInput}
```

Comment:
so `rest` is typed in such a way that if we pass it `disabled` then it doesn't expect to have `updatePolicyInput`, and `InlineRadioGroup` always takes an `onChange` (like `EuiRadioGroup` that it uses), so we need to pass something. 

and we don't want to change the types and always pass `updatePolicyInput` because we don't update the inputs when we're editing

Reply:
fixed, passing `disabled` to `InlineRadioGroup`. 

removed the union type for `rest`